fvm/evm/stdlib/contract.cdc

```diff
@@ -815,4 +813,19 @@ contract EVM {
             ?.borrowBridgeAccessor()
             ?? panic("Could not borrow reference to the EVM bridge")
     }
+
+    /// The Heartbeat resource controls the block production 
+    access(all) resource Heartbeat {
+
+        /// heartbeat calls commit block proposals and forms new blocks including all the 
+        /// recently executed transactions.
+        /// The Flow protocol makes sure to call this function once per block as a system call. 
+        access(all) fun heartbeat() {
+            InternalEVM.commitBlockProposal()
+        }
+    }
+    
+    init() { 
+        self.account.storage.save(<-create Heartbeat(), to: /storage/EVMHeartbeat)
+    }
```

Comment:
This is supposed to expose a heartbeat for the system chunk and prevent block production for anyone else. Though its assumption is the EVM is deployed to service account (not the storage account)

Reply:
Got it, I'll add a separate method for it in the following PR.